classes/class-p4-campaigns.php

```diff
@@ -285,5 +293,137 @@ public function enqueue_admin_assets() {
 			wp_enqueue_script( $this->taxonomy );
 			wp_enqueue_media();
 		}
+
+		/**
+		 * Register campaigns cpt
+		 */
+		public function register_campaigns_cpt() {
+
+			$labels = array(
+				'name'               => _x( 'Campaigns', 'post type general name', 'planet4-master-theme-backend' ),
+				'singular_name'      => _x( 'Campaign', 'post type singular name', 'planet4-master-theme-backend' ),
+				'menu_name'          => _x( 'Campaigns', 'admin menu', 'planet4-master-theme-backend' ),
+				'name_admin_bar'     => _x( 'Campaign', 'add new on admin bar', 'planet4-master-theme-backend' ),
+				'add_new'            => _x( 'Add New', 'campaign', 'planet4-master-theme-backend' ),
+				'add_new_item'       => __( 'Add New Campaign', 'planet4-master-theme-backend' ),
+				'new_item'           => __( 'New Campaign', 'planet4-master-theme-backend' ),
+				'edit_item'          => __( 'Edit Campaign', 'planet4-master-theme-backend' ),
+				'view_item'          => __( 'View Campaign', 'planet4-master-theme-backend' ),
+				'all_items'          => __( 'All Campaigns', 'planet4-master-theme-backend' ),
+				'search_items'       => __( 'Search Campaigns', 'planet4-master-theme-backend' ),
+				'parent_item_colon'  => __( 'Parent Campaigns:', 'planet4-master-theme-backend' ),
+				'not_found'          => __( 'No campaigns found.', 'planet4-master-theme-backend' ),
+				'not_found_in_trash' => __( 'No campaigns found in Trash.', 'planet4-master-theme-backend' )
+			);
+
+			$args = array(
+				'labels'             => $labels,
+				'description'        => __( 'Campaigns', 'planet4-master-theme-backend' ),
+				'public'             => true,
+				'publicly_queryable' => true,
+				'show_ui'            => true,
+				'show_in_menu'       => true,
+				'query_var'          => true,
+				'rewrite'            => array( 'slug' => 'campaign' ),
+				'capability_type'    => 'post',
+				'has_archive'        => true,
+				'taxonomies'         => array( 'category', 'post_tag' ),
+				'hierarchical'       => false,
+				'menu_position'      => null,
+				'menu_icon'          => 'dashicons-megaphone',
+				'supports'           => array( 'title', 'editor', 'author', 'thumbnail', 'excerpt', 'comments' )
+			);
+
+			register_post_type( $this->post_type, $args );
+
+		}
+
+		/**
+		 * Add metabox for campaign page template selection on campaigns cpt
+		 */
+		public function campaign_page_templates_meta_box() {
+			add_meta_box(
+				'campaigns-page-templates',
+				__( 'Page Templates', 'sitepoint' ),
+				array( $this, 'campaign_page_templates_meta_box_callback' ),
+				'campaigns',
+				'side'
+			);
+		}
+
+		/**
+		 * callback function for campaign page template selection
+		 *
+		 * @param $post
+		 */
+		public function campaign_page_templates_meta_box_callback( $post ) {
+
+			// Add a nonce field so we can check for it later.
+			wp_nonce_field( 'campaign_page_template_nonce', 'campaign_page_template_nonce' );
```

Comment:
It's usually better to include something unique (e.g. 'edit_campaign-' . $post->ID) to this post in the nonce action. Otherwise, theoretically, this nonce could be used to modify a different post.